src/interfaces.js

```diff
@@ -132,5 +132,5 @@ interface ConfigureRouter {
   /**
   * Implement this hook if you want to add sub-routes to your view.
   */
-  configureRouter(config: RouterConfiguration, router: Router);
+  configureRouter(config: RouterConfiguration, router: Router): any;
```

Comment:
[router.js#L117](https://github.com/aurelia/router/blob/master/src/router.js#L117)
Like I said, I'm probably digging too deep and `void` probably is good enough.
Also in your sample the `configureRouter` is missing explicit return type.